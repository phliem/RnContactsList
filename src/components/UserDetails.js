// @flow
import React from 'react';
import { StyleSheet, Text, View } from 'react-native';

// Constants & Services
import type USER from '../res/types';
import COLORS from '../res/colors';
import TYPOGRAPHY from '../res/typography';
import SIZES from '../res/sizes';

type PROPS = {
  style: {},
  user: USER,
};

const UserDetails = (props: PROPS) => {
  return (
    <View style={[props.style, styles.wrap]}>
      <View style={styles.title}>
        <Text style={TYPOGRAPHY.h1}>
          {props.user.firstname}{' '}
          <Text style={styles.regular}>{props.user.lastname}</Text>
        </Text>
        <Text style={TYPOGRAPHY.h2}>{props.user.job}</Text>
      </View>
      <View style={styles.separator} />
      <Text style={TYPOGRAPHY.h3}>About me</Text>
      <Text style={TYPOGRAPHY.text}>
        Lorem ipsum dolor sit, amet consectetur adipisicing elit. Deleniti
        architecto hic optio illum rerum, nisi sunt accusantium in vero
        voluptate similique earum quidem possimus, dolores saepe nobis
        perspiciatis dignissimos reiciendis. At vero eos et accusamus et iusto
        odio dignissimos ducimus qui blanditiis praesentium voluptatum deleniti
        atque corrupti quos dolores et quas molestias excepturi sint occaecati
        cupiditate non provident, similique sunt in culpa qui officia deserunt
        mollitia animi, id est laborum et dolorum fuga. Et harum quidem rerum
        facilis est et expedita distinctio. Nam libero tempore, cum soluta nobis
        est eligendi optio cumque nihil impedit quo minus id quod maxime placeat
        facere possimus, omnis voluptas assumenda est.
      </Text>
    </View>
  );
};

const styles = StyleSheet.create({
  wrap: {
    backgroundColor: COLORS.backgroundMain,
    padding: SIZES.spaceM,
  },
  title: {
    marginTop: SIZES.spaceM,
    justifyContent: 'center',
    alignItems: 'center',
  },
  regular: {
    fontWeight: 'normal',
  },
  separator: {
    borderWidth: 0.5,
    borderColor: COLORS.light,
    margin: SIZES.spaceL,
  },
});

export default UserDetails;
