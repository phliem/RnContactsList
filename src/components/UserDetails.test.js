import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

// Components
import UserDetails from './UserDetails';

describe('Should test UserDetails Component', () => {
  test('render and match snapshot', () => {
    const props = {
      firstname: 'Liem',
      lastname: 'Pham',
      job: 'Developer',
    };

    const testRender = renderer.create(<UserDetails user={props} />);
    const testInstance = testRender.toJSON();
    expect(testInstance).toMatchSnapshot();
  });
});
