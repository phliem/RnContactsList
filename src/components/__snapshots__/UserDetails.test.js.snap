// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should test UserDetails Component render and match snapshot 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "padding": 20,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginTop": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 26,
          "fontWeight": "bold",
          "marginBottom": 5,
        }
      }
    >
      Liem
       
      <Text
        style={
          Object {
            "fontWeight": "normal",
          }
        }
      >
        Pham
      </Text>
    </Text>
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 22,
          "marginBottom": 5,
        }
      }
    >
      Developer
    </Text>
  </View>
  <View
    style={
      Object {
        "borderColor": "#ffffff",
        "borderWidth": 0.5,
        "margin": 40,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#ffffff",
        "fontSize": 20,
        "fontWeight": "bold",
        "marginBottom": 10,
      }
    }
  >
    About me
  </Text>
  <Text
    style={
      Object {
        "color": "#ffffff",
        "lineHeight": 20,
      }
    }
  >
    Lorem ipsum dolor sit, amet consectetur adipisicing elit. Deleniti architecto hic optio illum rerum, nisi sunt accusantium in vero voluptate similique earum quidem possimus, dolores saepe nobis perspiciatis dignissimos reiciendis. At vero eos et accusamus et iusto odio dignissimos ducimus qui blanditiis praesentium voluptatum deleniti atque corrupti quos dolores et quas molestias excepturi sint occaecati cupiditate non provident, similique sunt in culpa qui officia deserunt mollitia animi, id est laborum et dolorum fuga. Et harum quidem rerum facilis est et expedita distinctio. Nam libero tempore, cum soluta nobis est eligendi optio cumque nihil impedit quo minus id quod maxime placeat facere possimus, omnis voluptas assumenda est.
  </Text>
</View>
`;
